packages/labs/ssr/src/lib/dom-shim.ts

```diff
@@ -55,8 +55,10 @@ export const getWindow = ({
       old: string | null,
       value: string | null
     ): void;
-    setAttribute(name: string, value: string) {
-      attributesForElement(this).set(name, value);
+    setAttribute(name: string, value: unknown) {
+      // The browser will silently cast all values to strings. E.g. `42` becomes
+      // `"42"` and `{}` becomes `"[object Object]""`.
+      attributesForElement(this).set(name, String(value));
```

Comment:
btw, we found that `typeof value ==='string' ? value : String(value)` was measurably faster in lit-html.

Reply:
There are a whole load of spots where we do this, not sure I want to update them all? Are we really sure it's faster?  I don't actually see this pattern in https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts